source.go

```diff
@@ -60,7 +79,11 @@ type Source struct {
 }
 
 func NewSource() sdk.Source {
-	return &Source{}
+	return NewSourceWithClient(client.New())
+}
+
+func NewSourceWithClient(c Client) sdk.Source {
```

Comment:
If we change the return type to `*Source` we don't have to cast the object in `setupTest`.
```suggestion
func NewSourceWithClient(c Client) *Source {
```